src/lib/comms/include/sol-http.h

```diff
@@ -0,0 +1,211 @@
+/*
+ * This file is part of the Soletta Project
+ *
+ * Copyright (C) 2015 Intel Corporation. All rights reserved.
+ *
+ * Redistribution and use in source and binary forms, with or without
+ * modification, are permitted provided that the following conditions
+ * are met:
+ *
+ *   * Redistributions of source code must retain the above copyright
+ *     notice, this list of conditions and the following disclaimer.
+ *   * Redistributions in binary form must reproduce the above copyright
+ *     notice, this list of conditions and the following disclaimer in
+ *     the documentation and/or other materials provided with the
+ *     distribution.
+ *   * Neither the name of Intel Corporation nor the names of its
+ *     contributors may be used to endorse or promote products derived
+ *     from this software without specific prior written permission.
+ *
+ * THIS SOFTWARE IS PROVIDED BY THE COPYRIGHT HOLDERS AND CONTRIBUTORS
+ * "AS IS" AND ANY EXPRESS OR IMPLIED WARRANTIES, INCLUDING, BUT NOT
+ * LIMITED TO, THE IMPLIED WARRANTIES OF MERCHANTABILITY AND FITNESS FOR
+ * A PARTICULAR PURPOSE ARE DISCLAIMED. IN NO EVENT SHALL THE COPYRIGHT
+ * OWNER OR CONTRIBUTORS BE LIABLE FOR ANY DIRECT, INDIRECT, INCIDENTAL,
+ * SPECIAL, EXEMPLARY, OR CONSEQUENTIAL DAMAGES (INCLUDING, BUT NOT
+ * LIMITED TO, PROCUREMENT OF SUBSTITUTE GOODS OR SERVICES; LOSS OF USE,
+ * DATA, OR PROFITS; OR BUSINESS INTERRUPTION) HOWEVER CAUSED AND ON ANY
+ * THEORY OF LIABILITY, WHETHER IN CONTRACT, STRICT LIABILITY, OR TORT
+ * (INCLUDING NEGLIGENCE OR OTHERWISE) ARISING IN ANY WAY OUT OF THE USE
+ * OF THIS SOFTWARE, EVEN IF ADVISED OF THE POSSIBILITY OF SUCH DAMAGE.
+ */
+
+#pragma once
+
+#include <sol-buffer.h>
+#include <sol-vector.h>
+
+#ifdef __cplusplus
+extern "C" {
+#endif
+
+/**
+ * @defgroup HTTP
+ * @ingroup Comms
+ *
+ * @{
+ */
+
+enum sol_http_method {
+    SOL_HTTP_METHOD_GET,
+    SOL_HTTP_METHOD_POST,
+    SOL_HTTP_METHOD_HEAD,
+    SOL_HTTP_METHOD_INVALID
+};
+
+enum sol_http_param_type {
+    SOL_HTTP_PARAM_QUERY_PARAM,
+    SOL_HTTP_PARAM_COOKIE,
+    SOL_HTTP_PARAM_POST_FIELD,
+    SOL_HTTP_PARAM_HEADER,
+    SOL_HTTP_PARAM_AUTH_BASIC,
+    SOL_HTTP_PARAM_ALLOW_REDIR,
+    SOL_HTTP_PARAM_TIMEOUT,
+    SOL_HTTP_PARAM_VERBOSE
+};
+
+enum sol_http_status_code {
+    SOL_HTTP_STATUS_OK = 200,
+    SOL_HTTP_STATUS_NOT_MODIFIED = 304,
+    SOL_HTTP_STATUS_BAD_REQUEST = 400,
+    SOL_HTTP_STATUS_FORBIDDEN = 403,
+    SOL_HTTP_STATUS_NOT_FOUND = 404,
+    SOL_HTTP_STATUS_METHOD_NOT_ALLOWED = 405,
+    SOL_HTTP_STATUS_METHOD_INTERNAL_ERROR = 500,
```

Comment:
Missing link to https://http.cat/ and, of course, HTTP 418: https://http.cat/418